test/index.js

```diff
@@ -183,6 +183,14 @@ test('excess leading dashes on options are retained', function(t) {
 
 // Test bad inputs
 
+test('invalid argument passed for options', function(t) {
+  const passedArgs = ['--so=wat'];
+
+  t.throws(function() { parseArgs(passedArgs, 'bad value'); });
```

Comment:
It would be good to validate the type of error thrown here like you do below, unless there's a reason not to that I'm missing.